starknet_py/net/account/account.py

```diff
@@ -96,11 +86,15 @@ def __init__(
                     "Either a signer or a key_pair must be provided in Account constructor."
                 )
 
-            chain = chain_from_network(net=client.net, chain=chain)
             signer = StarkCurveSigner(
-                account_address=self.address, key_pair=key_pair, chain_id=chain
+                account_address=self.address,
+                key_pair=key_pair,
+                chain_id=cast(
+                    StarknetChainId, chain
+                ),  # cast is required, because pyright doesn't know chain is not None here
```

Comment:
How come it wasn't required before?

Reply:
`chain_from_network()` returns StarknetChainId or raises an error, so the chain was always StarknetChainId here

Reply:
We could try doing an `assert` 👀

Reply:
Casting in situations like these always requires an additional comment explaining the reason behind it.
I think
```
assert chain is not None
```
is shorter and self explanatory

Reply:
I know who would not like it 🙈

Reply:
😆
It wasn't about the use of `assert` though but rather the same `assert` in multiple places in code, implying a flaw in the structure of the code

Reply:
I'll leave it as open discussion 😅

Reply:
It is perfectly viable for `chain` to be `None` here, so i don't know why the `cast` is here - py**right** seems to be **right** 🙃

Reply:
I actually think that having this kind of implicit logic here makes for confusing code reading experience, did you guys consider changing it?

Reply:
**chain** cannot be `None` here, the check for it is few lines above:
```
        if chain is None and signer is None:
            raise ValueError("One of chain or signer must be provided.")
```

Reply:
I agree that the code is hard to follow though.
There was some attempt at making it more explicit in [this _closed_ PR](https://github.com/software-mansion/starknet.py/pull/736/files#diff-71fc6b5ba6c214b0ddf79941772b54fef34507113a848e666dbc8e9c1002a0f3).
Maybe it should be implemented in some other PR nonetheless.

Reply:
it is fixed now! Check it out